subprojects/junit5/src/main/java/org/mockito/internal/junit5/AnnotationUtil.java

```diff
@@ -0,0 +1,43 @@
+package org.mockito.internal.junit5;
+
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.mockito.junit5.Strictness;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.Optional;
+
+import static org.junit.platform.commons.support.AnnotationSupport.findAnnotation;
+import static org.mockito.quality.Strictness.WARN;
+
+public class AnnotationUtil {
+    /**
+     * FIXME we should use a more common place in mockito (MockitoCore?) to define the default strictness,
+     * it should  be uses by Runner & Rule too
+     */
+    private final static org.mockito.quality.Strictness DEFAULT_STRICTNESS = WARN;
+
+    public static org.mockito.quality.Strictness retrieveStrictness(ExtensionContext context) {
```

Comment:
If I understand correctly, this method is recursively looking up the annotation. I think we should approach this issue differently:

1. Nested classes can specify a new runner which overrides the parent runner.
2. The `@ExtendWith` specifies the strictness as well and this strictness will be used for all tests. E.g. `@ExtendWith(MockitoExtension.Silent.class)` or `@ExtendWith(MockitoExtension.Strict.class)`. `@ExtendWith(MockitoExtension.class)` is equivalent to the strict variant.
3. The user can opt-in to a different strictness per `method`.

The algorithm to retrieve the strictness is then:

1. If Method has annotation, use that strictness
2. Else use strictness specified by runner

As every runner already defines it strictness level, we don't need to fallback to the `DEFAULT_STRICTNESS`. Moreover, we can remove this field as a whole, as we are now use runners instead that have it baked in.

Reply:
@TimvdLippe 

Defining a strictness-level is from my working experience a rarely used feature, the most teams tend to move fast forward and use the default strictness. Therefore I tend to provide an API that is as simple as possible to cover 80% of the uses cases. That means a plain annotation like `@WithMockito` instead of @ExtendWith(MockitoExtension.Silent.class) to avoid boilerplate code. Readability is a also to consider here. 
When users want to define the strictness-level one concept of defining it should be enough, different concepts may be a bit uncomfortable and harder to remember.

Lets take the discussion to the google group -> https://groups.google.com/forum/#!topic/mockito/R1L1aG2RgBQ